src/Tribe/Plugins_API.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+class Tribe__Plugins_API {
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @since tbd
+	 *
+	 * @return Tribe__Plugins_API
+	 */
+	public static function instance() {
+		return tribe( 'plugins.api' );
+	}
+
+	/**
+	 * Setup hooks
+	 *
+	 * @since tbd
+	 */
+	public function hook() {
+		add_filter( 'plugins_api_result', array( $this, 'filter_api_result' ), 10, 3 );
+		add_filter( 'plugin_install_action_links', array( $this, 'filter_action_links' ), 10, 2 );
+	}
+
+	/**
+	 * Filters the action links for the plugin install page
+	 *
+	 * @since tbd
+	 *
+	 * @param array $links  Links for plugin
+	 * @param array $plugin Plugin data
+	 *
+	 * @return array
+	 */
+	public function filter_action_links( $links, $plugin ) {
+		if ( empty( $plugin['tribe-result'] ) ) {
+			return $links;
+		}
+
+		if ( ! empty( $plugin['buy-now'] ) ) {
+			// remove "more details" link from non-public plugins
+			array_pop( $links );
+
+			$link = '<a class="%s" target="_blank" data-slug="' . esc_attr( $plugin['slug'] ) . '" href="%s" aria-label="%s" data-name="' . esc_attr( $plugin['name'] ) . '">%s</a>';
+
+			if ( ! empty( $links[0] ) && preg_match( '/install-now/', $links[0] ) ) {
+				$links[0] = sprintf(
+					$link,
+					'button',
+					esc_url( $plugin['buy-now'] ),
+					esc_attr( sprintf( __( 'Buy %s now', 'tribe-common' ), $plugin['name'] ) ),
+					esc_html( __( 'Buy Now', 'tribe-common' ) )
```

Comment:
not a blocker but any reason using `esc_html( __() )` instead of just `esc_html__()` ?